### PR TITLE
chore(version): Bump version

### PR DESCRIPTION
**Pull request by dnys1**
### Features
- feat(auth): TOTP MFA
- feat(authenticator): TOTP MFA ([https://github.com/aws-amplify/amplify-flutter/pull/3609](https://github.com/aws-amplify/amplify-flutter/pull/3609))

### Fixes
- fix(analytics): AWSPinpointUserProfile Added null check for user attributes ([https://github.com/aws-amplify/amplify-flutter/pull/3598](https://github.com/aws-amplify/amplify-flutter/pull/3598))
- fix(auth): SDK exception mapping
- fix(authenticator): Social button debug Property Type ([https://github.com/aws-amplify/amplify-flutter/pull/3608](https://github.com/aws-amplify/amplify-flutter/pull/3608))
- fix(datastore): Use platform thread ([https://github.com/aws-amplify/amplify-flutter/pull/3607](https://github.com/aws-amplify/amplify-flutter/pull/3607))

Updated-Components: amplify_lints, Amplify Flutter, Amplify Dart, Amplify UI, DB Common, Secure Storage, AWS Common, Smithy, Worker Bee

### CODE REVIEW

File: packages/amplify/amplify_flutter/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+## 1.4.0
+
+### Features
+- feat(auth): TOTP MFA
+- feat(authenticator): TOTP MFA ([#3609](https://github.com/aws-amplify/amplify-flutter/pull/3609))
+
+### Fixes
+- fix(datastore): Use platform thread ([#3607](https://github.com/aws-amplify/amplify-flutter/pull/3607))
```
Comment (NikaHsn) [inline packages/amplify/amplify_flutter/CHANGELOG.md:8]: should other fixes be here too?

File: packages/amplify_core/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,12 @@
+## 1.4.0
+
+### Features
+- feat(auth): TOTP MFA
+
+### Fixes
+- fix(analytics): AWSPinpointUserProfile Added null check for user attributes ([#3598](https://github.com/aws-amplify/amplify-flutter/pull/3598))
+- fix(auth): SDK exception mapping
```
Comment (NikaHsn) [inline packages/amplify_core/CHANGELOG.md:8]: why do we have analytics here?

File: packages/api/amplify_api/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
+## 1.4.0
+
+### Features
+- feat(authenticator): TOTP MFA ([#3609](https://github.com/aws-amplify/amplify-flutter/pull/3609))
```
Comment (NikaHsn) [inline packages/api/amplify_api/CHANGELOG.md:4]: why do we have authenticator here?

File: packages/storage/amplify_storage_s3/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,8 @@
+## 1.4.0
+
+### Features
+- feat(authenticator): TOTP MFA ([#3609](https://github.com/aws-amplify/amplify-flutter/pull/3609))
```
Comment (NikaHsn) [inline packages/storage/amplify_storage_s3/CHANGELOG.md:4]: should this be here?

File: packages/amplify_core/CHANGELOG.md (hunk 2 above)
Comment (dnys1) [reply packages/amplify_core/CHANGELOG.md:8]: Core+Flutter package aggregate all the changes

**Review by Equartey — APPROVED** [review]

**Review by Jordan-Nelson — APPROVED** [review]

File: packages/amplify_core/CHANGELOG.md (hunk 2 above)
Comment (NikaHsn) [reply packages/amplify_core/CHANGELOG.md:8]: if so there are missing some missing feat and fix

File: packages/auth/amplify_auth_cognito/CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,9 @@
+## 1.4.0
+
+### Features
+- feat(auth): TOTP MFA
+- feat(authenticator): TOTP MFA ([#3609](https://github.com/aws-amplify/amplify-flutter/pull/3609))
```
Comment (NikaHsn) [inline packages/auth/amplify_auth_cognito/CHANGELOG.md:5]: not sure if feat(authenticator) should be here, the pr does not have any change in the auth package

File: packages/secure_storage/amplify_secure_storage_test/pubspec.yaml [hunk 6]
```diff
@@ -8,8 +8,7 @@ environment:
 
 dependencies:
   test: ^1.22.1
-  amplify_secure_storage_dart:
-    path: ../amplify_secure_storage_dart
+  amplify_secure_storage_dart: any
```
Comment (NikaHsn) [inline packages/secure_storage/amplify_secure_storage_test/pubspec.yaml:11]: do we need an override to get it from local path?

File: packages/smithy/smithy_codegen/pubspec.yaml [hunk 7]
```diff
@@ -9,8 +9,8 @@ environment:
 
 dependencies:
   args: ^2.3.0
-  aws_common: ">=0.5.0 <0.6.0"
-  aws_signature_v4: ">=0.4.0 <0.5.0"
+  aws_common: any
+  aws_signature_v4: any
```
Comment (NikaHsn) [inline packages/smithy/smithy_codegen/pubspec.yaml:13]: not sure what is this change for?

File: packages/smithy/smithy_test/pubspec.yaml [hunk 8]
```diff
@@ -9,13 +9,13 @@ environment:
 
 dependencies:
   async: ^2.10.0
-  aws_common: ">=0.4.0 <0.5.0"
-  aws_signature_v4: ">=0.3.1+2 <0.4.0"
+  aws_common: any
+  aws_signature_v4: any
```
Comment (NikaHsn) [inline packages/smithy/smithy_test/pubspec.yaml:13]: ditto

File: packages/smithy/smithy_test/pubspec.yaml [hunk 9]
```diff
@@ -9,13 +9,13 @@ environment:
...
   built_value: ">=8.6.0 <8.7.0"
   collection: ^1.15.0
   fixnum: ^1.0.0
   path: ^1.8.0
-  smithy: ">=0.4.0+1 <0.5.0"
+  smithy: any
```
Comment (NikaHsn) [inline packages/smithy/smithy_test/pubspec.yaml:18]: ditto

File: packages/amplify_core/CHANGELOG.md (hunk 2 above)
Comment (dnys1) [reply packages/amplify_core/CHANGELOG.md:8]: I should've been more clear. Core captures all the Dart-only changes, Flutter captures all the Dart+Flutter changes.

File: packages/auth/amplify_auth_cognito/CHANGELOG.md (hunk 5 above)
Comment (dnys1) [reply packages/auth/amplify_auth_cognito/CHANGELOG.md:5]: It did have a small change, but yeah I could have removed this.

File: packages/smithy/smithy_codegen/pubspec.yaml (hunk 7 above)
Comment (dnys1) [reply packages/smithy/smithy_codegen/pubspec.yaml:13]: It's from this: https://github.com/aws-amplify/amplify-flutter/pull/3568

File: packages/secure_storage/amplify_secure_storage_test/pubspec.yaml (hunk 6 above)
Comment (dnys1) [reply packages/secure_storage/amplify_secure_storage_test/pubspec.yaml:11]: Yes. That is handled by `aft link`. Hardcoded paths can cause issues when publishing due to pub's constraint solver. See: https://github.com/aws-amplify/amplify-flutter/pull/3568

**Review by NikaHsn — APPROVED** [review]

**Pull request merged**